### PR TITLE
Specs for ruby-mpd.rb. Just #callback_thread left. 98.58% coverage.

### PR DESCRIPTION
**Pull request by whomwah**
Testing `callback_thread` is proving complex. I think the method needs
refactoring first to try and move some of the logic elsewhere to try and
make it easier to test.  I'll create another branch to try this out. I also found the way
`@socket` is set confusing. May try and work on that too in another branch for review.

### CODE REVIEW

**Comment by archSeer** [comment]
MPD supports a command called [`idle`](http://www.musicpd.org/doc/protocol/command_reference.html), which will stall the connection and report back when there is any change. One option would be opening a second connection to MPD and then waiting for an update on that socket.

I think it would be much more optimal as we get rid of polling every 100ms.

**Comment by whomwah** [comment]
```idle``` certainly looks interesting and worth exploring. I see that it only lets you know there has been a change in a subsystem and does not let you know what that change is. It's up to you to find that out. All thought I guess a simple ```mpd.status``` would be enough.

**Comment by archSeer** [comment]
Yeah, that's the idea. Any time idle returns, just fetch `mpd.status`.

File: lib/ruby-mpd.rb [hunk 1]
```diff
@@ -140,6 +140,8 @@ def disconnect
       @socket.close
     rescue Errno::EPIPE
       # socket was forcefully closed
+    ensure
+      @socket = nil
```
Comment (archSeer) [inline lib/ruby-mpd.rb]: I think you should just move the `reset_vars` call to here. It will clear out the variables for you.

**Pull request closed without merging**